### PR TITLE
feat: add `showConstraintMessages` option to `ValidationError.toString()`

### LINKED ISSUE #1611: Provide an option to ValidationError.toString to use constraint messages instead of names only
### Description
The `toString` funtion of the `ValidationError` lists all failed constraints, but uses the constraint name which does not give a hint on why the constraint failed. The constraint message would be much more meaningful and provide more information.

E.g. current default output for `ValidationError.toString`
```
An instance of RootClass has failed the validation:
 - property title has failed the following constraints: minLength, isString
```

What I would like it to be:
```
An instance of RootClass has failed the validation:
 - property title has failed the following constraints: title must be longer than or equal to 15 characters, title must be a string
```

### Proposed solution
Add an optional argument `useConstraintMessages` to the `ValidationError.toString` function that switches from constraint names to constraint messages as output.

### PR DESCRIPTION
**Pull request by ahoisl**
## Description
Add `useConstraintMessages` option parameter to ValidationError.toString function to use constraint messages instead of constraint names in the string representation for an error.

## Checklist

- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  - `npm run prettier:check` passes
  - `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

### Fixes

fixes #1611

### CODE REVIEW

**Comment by NoNameProvided** [comment]
Hi! Thanks for opening this PR, can you rebase or cherry-pick it to the latest `develop`?

**Comment by ahoisl** [comment]
> Hi! Thanks for opening this PR, can you rebase or cherry-pick it to the latest `develop`?

@NoNameProvided  I just cherry-picked the commit to the latest develop branch 👍

**Comment by ahoisl** [comment]
@NoNameProvided and I fixed the tests just now 😄

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
   * @param showConstraintMessages show constraint messages instead of constraint names
```

Comment (braaar) [reply src/validation/ValidationError.ts:48]: I think the word "use" is a bit unclear here.

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
    showConstraintMessages: boolean = false
```

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
      (showConstraintMessages ? Object.values : Object.keys)(this.constraints ?? {}).join(`, `);
```

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
              .map(childError => childError.toString(shouldDecorate, true, this.property, showConstraintMessages))
```

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
                childError.toString(shouldDecorate, true, `${parentPath}${formattedProperty}`, showConstraintMessages)
```

**Review by braaar — CHANGES_REQUESTED** [review]
I think "show" is a better word than "use". Other possible candidates are "print", "list", "return", "display".

File: src/validation/ValidationError.ts
Comment (ahoisl) [reply src/validation/ValidationError.ts]: I'm fine with that 😄

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts]:
```suggestion
   * @param showConstraintMessages show constraint messages instead of constraint names
```

File: src/validation/ValidationError.ts
Comment (braaar) [inline src/validation/ValidationError.ts:59]: Why is this null coalescing operation introduced?

`this.constraints ?? {}` replaces `this.constraints` in the `Object.keys`/`Object.values` call.

Did Typescript complain about this? In that case this would be an existing bug in the current code, no?

Comment (ahoisl) [reply src/validation/ValidationError.ts:59]: `this.constraints` can be undefined, TS was also complaining in the original code. The function is not called in any case where constraints is actually undefined, but this was the IMHO more readable alternative than passing the constraints as arguments

Comment (braaar) [reply src/validation/ValidationError.ts:59]: I see, so this PR is technically not just a feature PR, but it's fine to bake in the fix here, I think.

Comment (ahoisl) [reply src/validation/ValidationError.ts:59]: Fine with me

**Review by braaar — DISMISSED** [review]

**Comment by braaar** [comment]
Let's see what the workflow says

**Comment by ahoisl** [comment]
> Let's see what the workflow says

Apparently needs approval from a maintainer... 
@NoNameProvided , I guess, your review is required. Not sure who else is a maintainer? Thank you!

**Review by NoNameProvided — APPROVED** [review]

**Pull request merged**
